core/src/aaa/main/game/CameraInputProcessor.java

```diff
@@ -0,0 +1,91 @@
+package aaa.main.game;
+
+import aaa.main.screens.MainScreen;
+import aaa.main.util.Constants;
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.Input;
+import com.badlogic.gdx.InputProcessor;
+import com.badlogic.gdx.graphics.OrthographicCamera;
+
+
+public class CameraInputProcessor implements InputProcessor {
+    private OrthographicCamera camera;
+
+    public CameraInputProcessor(OrthographicCamera camera) {
+        this.camera = camera;
+    }
+
+    @Override
+    public boolean keyDown(int keycode) {
+
+
+        //Camera zoom controls
+        if (keycode == Input.Keys.MINUS && camera.zoom <= Constants.MAX_ZOOM) {
+            //zoom out
+            camera.zoom += 0.02f;
+        } else if (keycode == Input.Keys.EQUALS && camera.zoom >= Constants.MIN_ZOOM) {
+            //zoom in
+            camera.zoom -= 0.02f;
+        }
+
+        //move camera based on arrow keys (doesnt work when set to ortho)
+        //test msg
+        /*
+        if (keycode == Input.Keys.LEFT) {
+            //move left
+            camera.translate(-32, 0);
+        } else if (keycode == Input.Keys.RIGHT) {
+            //move right
+            camera.translate(32, 0);
+        } else if (keycode == Input.Keys.UP) {
+            //move up
+            camera.translate(0, 32);
+        } else if (keycode == Input.Keys.DOWN) {
+            //move down
+            camera.translate(0, -32);
+        }
+        */
+
+        return false;
+    }
+
+    @Override
+    public boolean keyUp(int keycode) {
+        return false;
```

Comment:
you do not need to @ Override all of these, any that you are not changing you should just be able to omit